pkg/bundle/create/create.go

```diff
@@ -12,45 +16,56 @@ import (
 )
 
 // CreateFull performs all tasks in creating full imagesets
-func CreateFull(rootDir string, dryRun, insecure bool) error {
+func CreateFull(configPath, rootDir string, dryRun, insecure bool) error {
+
 	ctx := context.Background()
 
+	sourceDir := filepath.Join(rootDir, config.SourcePath)
+	metadataPath := filepath.Join(sourceDir, "publish", config.MetadataFile)
+
 	err := bundle.MakeCreateDirs(rootDir)
 	if err != nil {
 		return err
 	}
-	// Open Metadata
+
+	// Read in current metadata
 	metadata, err := config.LoadMetadata(rootDir)
 	if err != nil {
 		return err
 	}
 
-	// TODO: this isn't the best way to handle metadata vs. no metadata.
-	// Needs refactoring.
 	var lastRun v1alpha1.PastMirror
+	// If metdata is found throw an error, else create new metadata
+	// Need refactoring
 	if len(metadata.PastMirrors) != 0 {
-		lastRun = metadata.PastMirrors[len(metadata.PastMirrors)-1]
-		logrus.Debug(lastRun)
+		return &config.MetadataError{Path: metadataPath, Type: "full"}
```

Comment:
WDYT about adding a helper for convenience in situations like this?
```go
func NewFullMetadataError(fpath string) error {
    return &MetadataError{Path: fpath, Type: "full"}
}

func NewDiffMetadataError(fpath string) error {
    return &MetadataError{Path: fpath, Type: "diff"}
}
```

Reply:
Added these in and added them in the test as well.